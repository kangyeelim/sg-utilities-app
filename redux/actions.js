export const UPDATE_E_READING = 'UPDATE_E_READING'
export const DELETE_E_READING = 'DELETE_E_READING'
export const UPDATE_MARKED_DATE = 'UPDATE_MARKED_DATE'
export const FORMAT_MARKED_DATE = 'FORMAT_MARKED_DATE'
export const UPDATE_W_READING = 'UPDATE_W_READING'
export const DELETE_W_READING = 'DELETE_W_READING'
export const UPDATE_G_READING = 'UPDATE_G_READING'
export const DELETE_G_READING = 'DELETE_G_READING'
export const UPDATE_BILL = 'UPDATE_BILL'
export const DELETE_BILL = 'DELETE_BILL'
export const UPDATE_PHOTO = 'UPDATE_PHOTO'
export const DELETE_PHOTO = 'DELETE_PHOTOs'
export const DELETE_UNSAVED_PHOTOS = 'DELETE_UNSAVED_PHOTOS'
export const UPDATE_REMINDER = 'UPDATE_REMINDER'
export const DELETE_REMINDER = 'DELETE_REMINDER'
export const RESET = 'RESET'
// action creators
export const updateElectReading = update => ({
  type: UPDATE_E_READING,
  payload: update,
})

export const deleteElectReading = reading => ({
  type: DELETE_E_READING,
  payload: reading,
})

export const updateMarkedDate = date => ({
  type: UPDATE_MARKED_DATE,
  payload: date,
})

export const updateWaterReading = update => ({
  type: UPDATE_W_READING,
  payload: update,
})

export const deleteWaterReading = reading => ({
  type: DELETE_W_READING,
  payload: reading,
})

export const updateGasReading = update => ({
  type: UPDATE_G_READING,
  payload: update,
})

export const deleteGasReading = reading => ({
  type: DELETE_G_READING,
  payload: reading,
})

export const updateBill = bill => ({
  type: UPDATE_BILL,
  payload: bill,
})

export const deleteBill = bill => ({
  type: DELETE_BILL,
  payload: bill,
})

export const updatePhoto = photo => ({
  type: UPDATE_PHOTO,
  payload: photo,
})

export const deletePhoto = photo => ({
  type: DELETE_PHOTO,
  payload: photo,
})

export const deleteUnsavedPhotos = () => ({
  type: DELETE_UNSAVED_PHOTOS
})

export const updateReminder = time => ({
  type: UPDATE_REMINDER,
  payload: time
})

export const deleteReminder = type => ({
  type: DELETE_REMINDER,
  payload: type
})

export const resetStore = () => ({
  type: RESET
})
