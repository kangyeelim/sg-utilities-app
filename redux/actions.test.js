import * as actions from './actions'

//from ReadingForm.js:
const electricity = {key:'electricity', color: 'red'};
const water = {key:'water', color: 'blue'};
const gas = {key:'gas', color: 'green'};

//from SettingsScreen.js
const DAILY_READING_REMINDER = 'daily_reading_reminder';

describe('updateElectReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.updateElectReading({id: '2020-01-01', reading: '100'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateElectReading({})).toMatchSnapshot()
  })

  it('handles empty reading', () => {
    expect(actions.updateElectReading({reading: ''})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.updateElectReading({id: ''})).toMatchSnapshot()
  })
})

describe('updateWaterReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.updateWaterReading({id: '2020-01-01', reading: '100'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateWaterReading({})).toMatchSnapshot()
  })

  it('handles empty reading', () => {
    expect(actions.updateWaterReading({reading: ''})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.updateWaterReading({id: ''})).toMatchSnapshot()
  })
})

describe('updateGasReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.updateGasReading({id: '2020-01-01', reading: '100'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateGasReading({})).toMatchSnapshot()
  })

  it('handles empty reading', () => {
    expect(actions.updateGasReading({reading: ''})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.updateGasReading({id: ''})).toMatchSnapshot()
  })
})

describe('updateMarkedDate returns actions', () => {
  it('returns an action', () => {
    expect(actions.updateMarkedDate({date: '2020-01-01', dots: [electricity]})).toMatchSnapshot()
    expect(actions.updateMarkedDate({date: '2020-01-01', dots: [electricity, water]})).toMatchSnapshot()
    expect(actions.updateMarkedDate({date: '2020-01-01', dots: [electricity, water, gas]})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateMarkedDate({})).toMatchSnapshot()
  })

  it('handles empty dots', () => {
    expect(actions.updateMarkedDate({dots: []})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.updateGasReading({date: ''})).toMatchSnapshot()
  })
})

describe('updateBill returns actions', () => {
  it('returns an action', () => {
    expect(actions.updateBill({month:'10', year:'2020', captures:['directory/test/jpg']})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateBill({})).toMatchSnapshot()
  })

  it('handles empty month', () => {
    expect(actions.updateBill({month: []})).toMatchSnapshot()
  })

  it('handles empty year', () => {
    expect(actions.updateBill({year: ''})).toMatchSnapshot()
  })

  it('handles empty captures', () => {
    expect(actions.updateBill({captures: []})).toMatchSnapshot()
  })
})

describe('updatePhoto returns actions', () => {
  it('returns an action', () => {
    expect(actions.updatePhoto({uri:'directory/test.jpg', saved:false})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updatePhoto({})).toMatchSnapshot()
  })

  it('handles empty uri', () => {
    expect(actions.updatePhoto({uri: ''})).toMatchSnapshot()
  })

  it('handles empty saved', () => {
    expect(actions.updatePhoto({saved: ''})).toMatchSnapshot()
  })

  it('handles empty captures', () => {
    expect(actions.updateBill({captures: []})).toMatchSnapshot()
  })
})

describe('updateReminder returns actions', () => {
  it('returns an action', () => {
    var date = new Date('2020-01-01');
    expect(actions.updateReminder({type:DAILY_READING_REMINDER, date:date, notification_id:'12345678'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.updateReminder({})).toMatchSnapshot()
  })

  it('handles empty type', () => {
    expect(actions.updateReminder({type: ''})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.updateReminder({date: ''})).toMatchSnapshot()
  })

  it('handles empty notification id', () => {
    expect(actions.updateReminder({notification_id: ''})).toMatchSnapshot()
  })
})

describe('resetStore returns actions', () => {
  it('returns an action', () => {
    expect(actions.resetStore()).toMatchSnapshot()
  })
})

describe('deleteUnsavedPhotos returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteUnsavedPhotos()).toMatchSnapshot()
  })
})

describe('deleteElectReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteElectReading({id:'2020-01-01'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deleteElectReading({})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.deleteElectReading({id: ''})).toMatchSnapshot()
  })
})

describe('deleteWaterReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteWaterReading({id:'2020-01-01'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deleteWaterReading({})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.deleteWaterReading({id: ''})).toMatchSnapshot()
  })
})

describe('deleteGasReading returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteGasReading({id:'2020-01-01'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deleteGasReading({})).toMatchSnapshot()
  })

  it('handles empty date', () => {
    expect(actions.deleteGasReading({id: ''})).toMatchSnapshot()
  })
})

describe('deletePhoto returns actions', () => {
  it('returns an action', () => {
    expect(actions.deletePhoto({uri:'directory/test.jpg', saved: false})).toMatchSnapshot()
    expect(actions.deletePhoto({uri:'directory/test.jpg', saved: true})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deletePhoto({})).toMatchSnapshot()
  })

  it('handles empty uri', () => {
    expect(actions.deletePhoto({uri: ''})).toMatchSnapshot()
  })

  it('handles empty saved', () => {
    expect(actions.deletePhoto({saved: ''})).toMatchSnapshot()
  })
})

describe('deleteBill returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteBill({month:'03', year: '2020'})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deleteBill({})).toMatchSnapshot()
  })

  it('handles empty month, no year', () => {
    expect(actions.deleteBill({month: ''})).toMatchSnapshot()
  })

  it('handles empty year, no month', () => {
    expect(actions.deleteBill({year: ''})).toMatchSnapshot()
  })
})

describe('deleteReminder returns actions', () => {
  it('returns an action', () => {
    expect(actions.deleteReminder({type:DAILY_READING_REMINDER})).toMatchSnapshot()
  })

  it('handles empty object', () => {
    expect(actions.deleteReminder({})).toMatchSnapshot()
  })

  it('handles empty type', () => {
    expect(actions.deleteReminder({type: ''})).toMatchSnapshot()
  })
})
