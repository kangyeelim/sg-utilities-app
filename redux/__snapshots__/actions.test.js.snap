// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteBill returns actions handles empty month, no year 1`] = `
Object {
  "payload": Object {
    "month": "",
  },
  "type": "DELETE_BILL",
}
`;

exports[`deleteBill returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_BILL",
}
`;

exports[`deleteBill returns actions handles empty year, no month 1`] = `
Object {
  "payload": Object {
    "year": "",
  },
  "type": "DELETE_BILL",
}
`;

exports[`deleteBill returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "month": "03",
    "year": "2020",
  },
  "type": "DELETE_BILL",
}
`;

exports[`deleteElectReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "DELETE_E_READING",
}
`;

exports[`deleteElectReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_E_READING",
}
`;

exports[`deleteElectReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
  },
  "type": "DELETE_E_READING",
}
`;

exports[`deleteGasReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "DELETE_G_READING",
}
`;

exports[`deleteGasReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_G_READING",
}
`;

exports[`deleteGasReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
  },
  "type": "DELETE_G_READING",
}
`;

exports[`deletePhoto returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_PHOTOs",
}
`;

exports[`deletePhoto returns actions handles empty saved 1`] = `
Object {
  "payload": Object {
    "saved": "",
  },
  "type": "DELETE_PHOTOs",
}
`;

exports[`deletePhoto returns actions handles empty uri 1`] = `
Object {
  "payload": Object {
    "uri": "",
  },
  "type": "DELETE_PHOTOs",
}
`;

exports[`deletePhoto returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "saved": false,
    "uri": "directory/test.jpg",
  },
  "type": "DELETE_PHOTOs",
}
`;

exports[`deletePhoto returns actions returns an action 2`] = `
Object {
  "payload": Object {
    "saved": true,
    "uri": "directory/test.jpg",
  },
  "type": "DELETE_PHOTOs",
}
`;

exports[`deleteReminder returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_REMINDER",
}
`;

exports[`deleteReminder returns actions handles empty type 1`] = `
Object {
  "payload": Object {
    "type": "",
  },
  "type": "DELETE_REMINDER",
}
`;

exports[`deleteReminder returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "type": "daily_reading_reminder",
  },
  "type": "DELETE_REMINDER",
}
`;

exports[`deleteUnsavedPhotos returns actions returns an action 1`] = `
Object {
  "type": "DELETE_UNSAVED_PHOTOS",
}
`;

exports[`deleteWaterReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "DELETE_W_READING",
}
`;

exports[`deleteWaterReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_W_READING",
}
`;

exports[`deleteWaterReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
  },
  "type": "DELETE_W_READING",
}
`;

exports[`resetStore returns actions returns an action 1`] = `
Object {
  "type": "RESET",
}
`;

exports[`updateBill returns actions handles empty captures 1`] = `
Object {
  "payload": Object {
    "captures": Array [],
  },
  "type": "UPDATE_BILL",
}
`;

exports[`updateBill returns actions handles empty month 1`] = `
Object {
  "payload": Object {
    "month": Array [],
  },
  "type": "UPDATE_BILL",
}
`;

exports[`updateBill returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_BILL",
}
`;

exports[`updateBill returns actions handles empty year 1`] = `
Object {
  "payload": Object {
    "year": "",
  },
  "type": "UPDATE_BILL",
}
`;

exports[`updateBill returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "captures": Array [
      "directory/test/jpg",
    ],
    "month": "10",
    "year": "2020",
  },
  "type": "UPDATE_BILL",
}
`;

exports[`updateElectReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "UPDATE_E_READING",
}
`;

exports[`updateElectReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_E_READING",
}
`;

exports[`updateElectReading returns actions handles empty reading 1`] = `
Object {
  "payload": Object {
    "reading": "",
  },
  "type": "UPDATE_E_READING",
}
`;

exports[`updateElectReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
    "reading": "100",
  },
  "type": "UPDATE_E_READING",
}
`;

exports[`updateGasReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "UPDATE_G_READING",
}
`;

exports[`updateGasReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_G_READING",
}
`;

exports[`updateGasReading returns actions handles empty reading 1`] = `
Object {
  "payload": Object {
    "reading": "",
  },
  "type": "UPDATE_G_READING",
}
`;

exports[`updateGasReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
    "reading": "100",
  },
  "type": "UPDATE_G_READING",
}
`;

exports[`updateMarkedDate returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "date": "",
  },
  "type": "UPDATE_G_READING",
}
`;

exports[`updateMarkedDate returns actions handles empty dots 1`] = `
Object {
  "payload": Object {
    "dots": Array [],
  },
  "type": "UPDATE_MARKED_DATE",
}
`;

exports[`updateMarkedDate returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_MARKED_DATE",
}
`;

exports[`updateMarkedDate returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "date": "2020-01-01",
    "dots": Array [
      Object {
        "color": "red",
        "key": "electricity",
      },
    ],
  },
  "type": "UPDATE_MARKED_DATE",
}
`;

exports[`updateMarkedDate returns actions returns an action 2`] = `
Object {
  "payload": Object {
    "date": "2020-01-01",
    "dots": Array [
      Object {
        "color": "red",
        "key": "electricity",
      },
      Object {
        "color": "blue",
        "key": "water",
      },
    ],
  },
  "type": "UPDATE_MARKED_DATE",
}
`;

exports[`updateMarkedDate returns actions returns an action 3`] = `
Object {
  "payload": Object {
    "date": "2020-01-01",
    "dots": Array [
      Object {
        "color": "red",
        "key": "electricity",
      },
      Object {
        "color": "blue",
        "key": "water",
      },
      Object {
        "color": "green",
        "key": "gas",
      },
    ],
  },
  "type": "UPDATE_MARKED_DATE",
}
`;

exports[`updatePhoto returns actions handles empty captures 1`] = `
Object {
  "payload": Object {
    "captures": Array [],
  },
  "type": "UPDATE_BILL",
}
`;

exports[`updatePhoto returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_PHOTO",
}
`;

exports[`updatePhoto returns actions handles empty saved 1`] = `
Object {
  "payload": Object {
    "saved": "",
  },
  "type": "UPDATE_PHOTO",
}
`;

exports[`updatePhoto returns actions handles empty uri 1`] = `
Object {
  "payload": Object {
    "uri": "",
  },
  "type": "UPDATE_PHOTO",
}
`;

exports[`updatePhoto returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "saved": false,
    "uri": "directory/test.jpg",
  },
  "type": "UPDATE_PHOTO",
}
`;

exports[`updateReminder returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "date": "",
  },
  "type": "UPDATE_REMINDER",
}
`;

exports[`updateReminder returns actions handles empty notification id 1`] = `
Object {
  "payload": Object {
    "notification_id": "",
  },
  "type": "UPDATE_REMINDER",
}
`;

exports[`updateReminder returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_REMINDER",
}
`;

exports[`updateReminder returns actions handles empty type 1`] = `
Object {
  "payload": Object {
    "type": "",
  },
  "type": "UPDATE_REMINDER",
}
`;

exports[`updateReminder returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "date": 2020-01-01T00:00:00.000Z,
    "notification_id": "12345678",
    "type": "daily_reading_reminder",
  },
  "type": "UPDATE_REMINDER",
}
`;

exports[`updateWaterReading returns actions handles empty date 1`] = `
Object {
  "payload": Object {
    "id": "",
  },
  "type": "UPDATE_W_READING",
}
`;

exports[`updateWaterReading returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_W_READING",
}
`;

exports[`updateWaterReading returns actions handles empty reading 1`] = `
Object {
  "payload": Object {
    "reading": "",
  },
  "type": "UPDATE_W_READING",
}
`;

exports[`updateWaterReading returns actions returns an action 1`] = `
Object {
  "payload": Object {
    "id": "2020-01-01",
    "reading": "100",
  },
  "type": "UPDATE_W_READING",
}
`;
